### ISSUE TITLE
[0.99] Does not handle vcf.bgz or non-gzipped vcfs

### ISSUE THREAD

**Issue by aofarrel** [aofarrel/TOPMed_Pipeline_First_Step_WDL#3]
Because of how cloud platforms work, we need to know the name of our output before it actually exists. If the input file can only be vcf.gz, then this works. But accounting for other possibilities will take some ingenuinity.

```
task runGds {
	input {
		File vcf
		String output_file_name = basename(sub(vcf, "\\.vcf.gz$", ".gds"))
	}
	
	command {
		set -eux -o pipefail
		echo "Calling R script vcfToGds.R"
		R --vanilla --args "~{vcf}" < /analysis_pipeline_WDL/R/vcf2gds.R
	}

	runtime {
		docker: "quay.io/aofarrel/topmed-pipeline-wdl:circleci-push"
	}

	output {
		File out = output_file_name
	}
}
```

We can't just return three different outputs where each one represents a possible file, because if those files never materialize we will error out.

We could return dummy files for the names that don't match, but would triple the number of output files and could result in operations being performed on dummy outputs. ...unless we made a task to drop those dummy outputs, but that feels ridiculous.

**Comment by aofarrel** [comment]
Currently untested on anything except vcf.gz, but this seems promising. It is inelegant, but I can live with that for now.

https://github.com/aofarrel/analysis_pipeline_WDL/tree/support-bcf-vcf-and-vcfbgz

**Comment by aofarrel** [comment]
If this solution works then this can probably close #1 too, as bcftools is already in the docker and iirc is called in the R script.

**Comment by aofarrel** [comment]
Merged: #11
